nixos/modules/services/torrent/qui.nix

```
@@ -0,0 +1,122 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (lib)
+    getExe
+    maintainers
+    mkEnableOption
+    mkIf
+    mkOption
+    mkPackageOption
+    ;
+  inherit (lib.types)
+    bool
+    path
+    port
+    str
+    submodule
+    ;
+  cfg = config.services.qui;
+  configFormat = pkgs.formats.toml { };
+  configFile = configFormat.generate "qui.toml" cfg.settings;
+in
+{
+  options = {
+    services.qui = {
+      enable = mkEnableOption "qui";
+
+      package = mkPackageOption pkgs "qui" { };
+
+      openFirewall = mkOption {
+        type = bool;
+        default = false;
+        description = "Open ports in the firewall for qui.";
+      };
+
+      secretFile = mkOption {
+        type = path;
```

```suggestion
        type = path;
        example = "/etc/secrets/qui.txt";
```

Example would be nice